blackjax/mcmc/hmc.py

```diff
@@ -210,17 +211,20 @@ def kernel(
         logdensity_fn: Callable,
         step_size: float,
         inverse_mass_matrix: Array,
+        **integration_steps_kwargs,
     ) -> tuple[DynamicHMCState, HMCInfo]:
         """Generate a new sample with the HMC kernel."""
-        num_integration_steps = integration_steps_fn(state.random_generator_arg)
+        num_integration_steps = integration_steps_fn(
+            state.random_generator_arg, **integration_steps_kwargs
+        )
         hmc_state = HMCState(state.position, state.logdensity, state.logdensity_grad)
         hmc_proposal, info = hmc_base(
             rng_key,
             hmc_state,
             logdensity_fn,
             step_size,
             inverse_mass_matrix,
-            num_integration_steps,
+            jax.numpy.asarray(num_integration_steps, dtype=int),
```

Comment:
I think it is better to emphasize `integration_steps_fn` should return `int`. Could you reverse this and add in the docstring instead?